okhttp/src/commonMain/kotlin/okhttp3/internal/-HttpUrlCommon.kt

```diff
@@ -602,7 +602,18 @@ internal object CommonHttpUrl {
   }
 
   internal fun HttpUrl.Builder.effectivePort(): Int {
-    return if (port != -1) port else HttpUrl.defaultPort(scheme!!)
+    if (port >= 1 && port <= 65535){
```

Comment:
I'm not sure I agree with this, why change a bad port like 65599 to 80/443?

Reply:
Shouldn't it fail immediately if the port is outside that range, and only use defaultPort for -1?

```
when (port) {
  -1 -> default port
  in 1..65535 -> port
  else -> throw
}
```

Reply:
But please add a test, so we can see it's fixing what you think is broken.  I'm worried if you can't reproduce it in a test.